apps/web/app/(app)/environments/[environmentId]/surveys/[surveyId]/edit/components/SurveyMenuBar.tsx

```diff
@@ -180,6 +185,11 @@ export const SurveyMenuBar = ({
     }
   };
 
+  const handleSaveAndGoBack = async () => {
+    await handleSurveySave(true);
+    router.back();
```

Comment:
`await handleSurveySave(true);` already navigates back because of the `true` parameter (which I dislike by the way because it's a different concerns and thus should be separated). But we only need one of these navigations.
Ideally remove the navigation part from the `handleSurveySave` function since it's confusing.